biz.aQute.bndlib/src/aQute/bnd/osgi/Verifier.java

```diff
@@ -673,7 +673,7 @@ public void verify() throws Exception {
 		verifyDirectives(Constants.IMPORT_PACKAGE, "resolution:", PACKAGEPATTERN, "package");
 		verifyDirectives(Constants.REQUIRE_BUNDLE, "visibility:|resolution:", SYMBOLICNAME, "bsn");
 		verifyDirectives(Constants.FRAGMENT_HOST, "extension:", SYMBOLICNAME, "bsn");
-		verifyDirectives(Constants.PROVIDE_CAPABILITY, "effective:|uses:", null, null);
+		verifyDirectives(Constants.PROVIDE_CAPABILITY, "effective:|register:|uses:", null, null);
```

Comment:
`register` is not a listed directive for all capabilities. See https://osgi.org/specification/osgi.core/7.0.0/framework.module.html#d0e2821. So it not quite correct here to allow in all cases. It is only specified for the "osgi.serviceloader" namespace: https://osgi.org/specification/osgi.cmpn/7.0.0/service.loader.html#org.osgi.service.serviceloader.ServiceLoaderNamespace.CAPABILITY_REGISTER_DIRECTIVE

So the allowance of register should be more specific.

Reply:
But we don't have a more specific verification for this header. the possible permutations are huge.

Reply:
There is one known capability namespace which uses the register directive. That does not seems like a large problem space.

Reply:
I'm talking about opening the can of worms of validating specific namespaces.

Reply:
True, but that is why things are the way they are. This change just suggests opening register up to every capability which is part way to simply allowing all possible directives. 

You could also use a -fixupmessage to elide the register warning.

Reply:
Imagine the developer uses a nice annotation `@ServiceProvider` You mean that he has to then write a fixupmessages to prevent the build from failing?